internal/lsp/uri/uri.go

```diff
@@ -19,6 +19,9 @@ func FromPath(client clients.Identifier, path string) string {
 		// Convert Windows path separators to Unix separators
 		path = filepath.ToSlash(path)
 
+		// spaces must be percent-encoded
+		path = strings.ReplaceAll(path, " ", "%20")
```

Comment:
This is the fix, going from paths to URIs when paths contained spaces was broken. This meant files were being reloaded from disk at incorrect times.

Reply:
Good find!